### PR TITLE
Create nested files/folders

### PR DESCRIPTION
**Pull request by yeion7**
## What kind of change does this PR introduce?

feature

## What is the current behavior?

You can't create nested files or folders

## What is the new behavior?

if you input a valid path you can create nested files and folders #2391

*flow*

create nested files

1. create optimistic folders
2. create optimistic file
3. upload file 
4. open file (in this way, we don't need to wait after upload the full folder structure)
5. upload folders
6. move file to folder

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. Create a nested folter structure
2. Create a nested file structure
3. Rename a file and set a new folder structure

## Checklist

- [ ] Documentation
- [ ] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/2932/commits/aece73891678010ed6a77ca4a85732add4cf1f46 is at https://pr2932.build.csb.dev/s/new.

**Comment by CompuIves** [comment]
WOW. THIS IS AWESOME!

**Comment by christianalfoni** [comment]
This certainly is impressive 😄 I suggest we wait bringing this in until the end of the week as the `vscodeeffect` branch touches a lot of these files. Just avoid having extra "tension" getting that in.

**Comment by arthurdenner** [comment]
@yeion7, could you solve the conflicts on this to see if we can get it merged, please?

Awesome work, btw!

**Comment by christianalfoni** [comment]
From discussion:

**christianalfoni**
*"Hmf... I have a really bad feeling about this nested directory stuff. We are throwing complexity on top of already way too complex stuff. The code becomes incredibly intricate, because:

- The directories evaluated from the filename might or might not contain new directories. It can also contain a mix of both. First part exists, but second needs to be created
- We have to introduce denormalize, which does not match our current optimistic logic
- A rename of a file can now also be a "move the file", but this move might be to a non existing directory, or partly non existing directory. The file might also be renamed and moved
- Reverting any failures is already quite complex and becomes just a mess with this added feature
- It will be so much easier to introduce this later, when all files and folders are based on paths. And maybe we will even use the VSCode explorer which has all this stuff built in

I just feel very strongly that this is going to bite us in the ass and become a distraction. The complexity cost is just too high with the current state of the code."*

In short summary this is an awesome PR 👍 😄 , but we are in the middle of changing out the concept of files and directories to reduce complexity and support new functionality. The complexity cost is just too high with the current state of the code. 

Feel really bad about closing this, but I hope our considerations came through. It is not you, it is us! 😂  This feature will be introduced shortly, but in a different form!

**Comment by yeion7** [comment]
@christianalfoni Thanks for the explanation, I was thinking in the same way about this PR 👍
